config/v1/types_infrastructure.go

```diff
@@ -101,6 +101,15 @@ type InfrastructureStatus struct {
 	// +kubebuilder:default=HighlyAvailable
 	// +kubebuilder:validation:Enum=HighlyAvailable;SingleReplica
 	InfrastructureTopology TopologyMode `json:"infrastructureTopology"`
+
+	// cpuPartitioning expresses that CPU partitioning is a currently enabled feature in the cluster.
+	// CPU Partitioning means that this cluster can support partitioning workloads to specific CPU Sets.
+	// Valid values are None and AllNodes. When omitted, the default value is None.
+	// +kubebuilder:default=None
+	// +default="None"
+	// +kubebuilder:validation:Enum=None;AllNodes
+	// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+	CPUPartitioning CPUPartitioningMode `json:"cpuPartitioning,omitempty"`
```

Comment:
This field should have an `optional` tag on it

Have you tested that the defaulting occurs before the enum validation?

Reply:
Added, and I have tested it via a running cluster, and via the integration tests. So it should default if no value is provided, and go through validation as expected, and of course, fail validation if some value is supplied. Updated take advantage of the integration tests.